ExampleVisionOS/EuclidMesh.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  EuclidMesh.swift
+//  ExampleVisionOS
+//
+//  Created by Hal Mueller on 3/5/24.
+//  Copyright © 2024 Nick Lockwood. All rights reserved.
+//
+
+import CoreGraphics
+import Euclid
+
+let euclidMesh: Mesh = {
+    let start = CFAbsoluteTimeGetCurrent()
+
+    // create some geometry using Euclid
+    let cube = Mesh.cube(size: 0.8, material: Color.red)
+    let sphere = Mesh.sphere(slices: 120, material: CGImage.checkerboard())
+    let mesh = cube.subtracting(sphere).makeWatertight()
```

Comment:
I know this PR is closed, but (and maybe this is the wrong place to ask this question), I'm curious how important `makeWatertight()` is here.

When running this on device as/is, the print of `Time: ` was ~4.5 seconds. If I remove the call to `makeWatertight()` and bump the sphere slices parameter down to 60, it runs in around .5 seconds, and there doesn't seem (to me) to be a discernible difference.

Reply:
@mgrider Are you sure it wasn't reducing the slices that had the main impact, not removing makeWatertight()? The latter is usually negligible.

In any case, the answer is that non-watertight shapes can cause weird artifacts when performing CSG operations, but in this case since it's the last step you can skip it.

Reply:
I almost forgot, if you are running in debug mode, switching to release mode (or just enabling optimized builds for the Euclid module in debug mode, which will still let you debug the example app) should yield a significant speedup.

Reply:
You're totally right. Adding `makeWatertight()` back yields: `Time: 0.6943490505218506`.

Reply:
And it's just my opinion, but I think a 4.5 second delay in the example project feels broken (or at least not especially great), whereas .5 gets a 🤷‍♂️. Of course there will be ways to optimize, but this is going to be what people see when they're evaluating Euclid as a dependency.

Reply:
Love that this got made/merged, BTW. Definitely aiming for (hopefully helpful) constructive criticism!

Reply:
There's still a problem with blocking the main thread. What I had missed earlier is that Apple's RK samples all use the `try! await Entity(named: "MyEntity")` idiom, calling the async method to load from a file. There is no async version of `init(_ mesh:...)`, so we should be calling that in a task and adding the content when it's ready. A real-world user of the framework may have very complicated/expensive meshes to build. Let me poke at that a bit and I'll post a new PR/branch.

Reply:
Everything in Euclid should be safe to run on a background thread. Distinct meshes should be safe to create in parallel on multiple threads/tasks as there's no global shared state.

Euclid already makes some use of multithreading internally which may not play nicely with async/await but is probably OK. Search for "batch" in the codebase if you're curious.

I plan to add async interfaces at some point, but it's complicated for various reasons. You're more than welcome to take a stab at it if you like.

Reply:
See what we think about https://github.com/nicklockwood/Euclid/pull/116. And let's continue conversation on that PR, close out this thread.